src/app/page.tsx

```diff
@@ -0,0 +1,45 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import styles from '@/page.module.scss';
+import toast from 'react-hot-toast';
+import NewsList from './components/NewsList/NewsList';
+import getNews from './utils/getNews';
+import { NewsWithId } from './models/types';
+
+type Props = {
+  searchParams: { q: string };
+};
+
+function Home({ searchParams }: Props) {
+  const [news, setNews] = useState<NewsWithId[] | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      if (searchParams.q) {
```

Comment:
You can use an earlier return to decrease block nesting. Like 
` if (!searchParams.q) return;`